### DISCUSSION TITLE
Expand minio deployment using existing hardware

### DISCUSSION THREAD

**Discussion by sparkblaze** [minio/minio#15016] [Q&A] [answered]
I initially deployed Minio as a 4x server, 12x disk distrubuted deployment with

`MINIO_VOLUMES="https://minio-p-fs{01...04}/minio/data{01...12}`

and have now reached the point where I need to extend the deployment. For reasons, I can't _currently_ expand horizontally, so I've attempted to expand vertically by adding another set of 12x disks to each host

When I update `MINIO_VOLUMES` to include the new disks, it just sits in a loop waiting for servers to be available/format the disk

`MINIO_VOLUMES="https://minio-p-fs{01...04}/minio/data{01...12} https://minio-p-fs{01...04}/minio/data{13...24}"`

> FS01: Waiting for all other servers to be online to format the disks
> FS02: Waiting for the first server to format the disks
> FS03: Waiting for the first server to format the disks
> FS04: Waiting for the first server to format the disks

Is it possible to expand Minio in this way?

**Comment by harshavardhana** [comment, marked as answer]
Yes it is possible make sure your drives can be written to by MinIO and it was correct permissions.

**Reply by sparkblaze** [reply]
Do I need to run an additional instance of Minio on the host to run the second pool? 

All of the new disks are writeable by the user, but they do throw the following error

> dial tcp 192.168.202.19:9000: connect: connection refused

which feels a tad weird as it only throws that for the new disks

**Reply by harshavardhana** [reply]
You need to specify the same command line options on all servers.

**Reply by sparkblaze** [reply]
What about when the same command line option is specified on all servers?

`MINIO_VOLUMES="https://minio-p-fs{01...04}/minio/data{01...12} https://minio-p-fs{01...04}/minio/data{13...24}"`

This prevents the new volumes from initializing, reverthing it back to the original `MINIO_VOLUMES="https://minio-p-fs{01...04}/minio/data{01...12}"` allows the original pool to start up. 

The volumes all have the same permissions applied.

Should I be defining `MINIO_VOLUMES` with repeated hosts, or as I'm trying to scale vertically, should I be adding the new volumes `/minio/data{13...24}` onto the end to have `/minio/data{01...24}`?


> Marking https://minio-p3-fs03.exe.nhs.uk:9000/minio/storage/minio/data19/v43 temporary offline; caused by Post "https://minio-p3-fs03.exe.nhs.uk:9000/minio/storage/minio/data19/v43/readall?disk-id=&file-path=format.json&volume=.minio.sys": dial tcp 192.168.202.19:9000: connect: connection refused

>Unable to read 'format.json' from https://minio-p3-fs02.exe.nhs.uk:9000/minio/data16: Do not upgrade one server at a time - please follow the recommended guidelines mentioned here https://github.com/minio/minio#upgrading-minio for your environment

I haven't upgraded minio
